### PR TITLE
Fix errors in Linux compilation

### PR DESCRIPTION
**Pull request by Tiline**

### CODE REVIEW

File: c_api/concurrentqueue.h [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+        #endif
+    #elif __linux__
+        #define MOODYCAMEL_EXPORT
+    #elif __APPLE__
+        #define MOODYCAMEL_EXPORT
+    #endif
 #else
-#define MOODYCAMEL_EXPORT
-#endif
+    #define MOODYCAMEL_EXPORT
```
Comment (cameron314) [inline c_api/concurrentqueue.h:22]: Doesn't make sense to redefine if already defined.

File: c_api/concurrentqueue.h [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+        #if defined(MOODYCAMEL_STATIC) //preferred way
+            #define MOODYCAMEL_EXPORT
+        #elif defined(DLL_EXPORT)
+            #define MOODYCAMEL_EXPORT __declspec(dllexport)
+        #else
+            #define MOODYCAMEL_EXPORT __declspec(dllimport)
+        #endif
+    #elif __linux__
+        #define MOODYCAMEL_EXPORT
+    #elif __APPLE__
```
Comment (cameron314) [inline c_api/concurrentqueue.h:18]: What about platforms besides Windows, Linux, and Apple?

File: c_api/concurrentqueue.h [hunk 3]
```diff
@@ -28,7 +30,6 @@ MOODYCAMEL_EXPORT int moodycamel_cq_create(MoodycamelCQHandle* handle);
 MOODYCAMEL_EXPORT int moodycamel_cq_destroy(MoodycamelCQHandle handle);
 MOODYCAMEL_EXPORT int moodycamel_cq_enqueue(MoodycamelCQHandle handle, MoodycamelValue value);
 MOODYCAMEL_EXPORT int moodycamel_cq_try_dequeue(MoodycamelCQHandle handle, MoodycamelValue* value);
-MOODYCAMEL_EXPORT size_t moodycamel_cq_size_approx(MoodycamelCQHandle handle);
```
Comment (cameron314) [inline c_api/concurrentqueue.h:31]: Still need this! (And the include for size_t too)

**Review by cameron314 — CHANGES_REQUESTED** [review]
What is the actual issue you're trying to solve? As far as I can tell the original code is correct.

**Comment by Tiline** [comment]
Sorry, I encountered the following error when trying to compile the project under Linux:
```
[build]# make
In file included from ../c_api/blockingconcurrentqueue.cpp:1:0:
../c_api/concurrentqueue.h:25:1: error: ‘MOODYCAMEL_EXPORT’ does not name a type
 MOODYCAMEL_EXPORT int moodycamel_cq_create(MoodycamelCQHandle* handle);
 ...
```
When I view that macros are undefined, they can only be defined under windows

I'm not sure if I need to judge other platforms???

**Comment by Tiline** [comment]
Sorry, I encountered the following error when trying to compile the project under Linux:
```
[build]# make
In file included from ../c_api/blockingconcurrentqueue.cpp:1:0:
../c_api/concurrentqueue.h:25:1: error: ‘MOODYCAMEL_EXPORT’ does not name a type
 MOODYCAMEL_EXPORT int moodycamel_cq_create(MoodycamelCQHandle* handle);
 ...
```
When I view that macros are undefined, they can only be defined under Windows

I'm not sure if I need to judge other platforms???

**Comment by cameron314** [comment]
> When I view that macros are undefined, they can only be defined under Windows

Where do you see this? In the original code, the `#ifdef _WIN32` is matched with an `#else #define MOODYCAMEL_EXPORT`.

Judging from the line number in your error message, it seems likely you're using an older version of the library which had this bug (fixed about a year ago: https://github.com/cameron314/concurrentqueue/commit/e6fec438e8639221d43dae4b2ddf133e20580fdd).

**Comment by Tiline** [comment]
对不起，我确认了下代码是在v1.0.3下，在master分支下编译正确

**Pull request closed without merging**